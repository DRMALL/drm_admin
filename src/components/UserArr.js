

import React from 'react'
import Table from 'antd/lib/table'
import Icon from 'antd/lib/icon'
import editUser from '../actions/editUser'
import userDel from '../actions/userDel'
import Modal from 'antd/lib/modal'
// import store from '../commons/store'

export default props => {
  const { userArr } = props.state.user
  const columns = [{
    title: '用户名称',
    dataIndex: 'name',
    key:'name'
  },{
    title: '用户邮箱',
    dataIndex: 'email',
    key:'email'
  },{
    title: '联系电话',
    dataIndex: 'phone',
    key:'phone'
  },{
    title: '公司名称',
    dataIndex: 'company_name',
    key:'company_name'
  },{
    title: '通讯地址',
    dataIndex: 'address',
    key:'address'
  },{
    title: '创建时间',
    dataIndex: 'time',
    key:'time'
  },{
    title: '操作',
    key:'action',
    render: (text, record) => (
      <div className='user-arr-action' >
        <div className='user-arr-item'  style={{ color: '#579df2' }}
             onClick={ editUser } data-user-id={ record._id } >
          <i className="iconfont icon-compile"  style={{ color: '#0068d2' }} ></i>
        </div>
        <div className='user-arr-item'  data-user-id={ record._id }
             onClick={ confirmDelUser } data-user-name={ record.name } ><Icon type="delete" /></div>
      </div>
      )
  }]
  return(
    <div>
      <Table columns={columns} dataSource={userArr} />
    </div>
    )
}



function confirmDelUser(e){
  const { userId, userName } = e.currentTarget.dataset
  Modal.confirm({
    title: '提示',
    content: `确认删除用户${userName},该操作不可逆!!`,
    onOk(){
      userDel(userId)
    },
    onCancle(){
      console.log('取消删除')
    },
    maskClosable: true
  })
}

