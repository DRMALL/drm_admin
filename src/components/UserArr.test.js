

import React from 'react'
import UserArr from './UserArr'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

it('UserArr test', () => {
  const state = { user: { userArr: [{ key:0, name: '张三', email: '3255243@fsaer.com' },{ key:1, name: '李四', email: 'ds55243@fsaer.com' }] } }
  const wrapper = render(<UserArr state={ state } />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
