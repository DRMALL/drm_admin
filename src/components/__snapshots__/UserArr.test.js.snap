// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserArr test 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-large ant-table-scroll-position-left"
        >
          <div
            class="ant-table-content"
          >
            <div
              class="ant-table-body"
            >
              <table
                class=""
              >
                <colgroup>
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:20%;min-width:20%;"
                  />
                  <col
                    style="width:20%;min-width:20%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class=""
                    >
                      <span>
                        用户名称
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        用户邮箱
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        联系电话
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        公司名称
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        通讯地址
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        创建时间
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        操作
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row  ant-table-row-level-0"
                  >
                    <td
                      class=""
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left:0px;"
                      />
                      张三
                    </td>
                    <td
                      class=""
                    >
                      3255243@fsaer.com
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div
                        class="table-icon-content"
                      >
                        <div
                          class="table-icon-item"
                          style="color:#579df2;"
                        >
                          <i
                            class="iconfont icon-compile"
                            style="color:#0068d2;font-size:25px;"
                          />
                        </div>
                        <div
                          class="table-icon-item"
                          data-user-name="张三"
                        >
                          <img
                            alt="删除"
                            src="dels.png"
                            style="width:20px;height:20px;"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row  ant-table-row-level-0"
                  >
                    <td
                      class=""
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left:0px;"
                      />
                      李四
                    </td>
                    <td
                      class=""
                    >
                      ds55243@fsaer.com
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div
                        class="table-icon-content"
                      >
                        <div
                          class="table-icon-item"
                          style="color:#579df2;"
                        >
                          <i
                            class="iconfont icon-compile"
                            style="color:#0068d2;font-size:25px;"
                          />
                        </div>
                        <div
                          class="table-icon-item"
                          data-user-name="李四"
                        >
                          <img
                            alt="删除"
                            src="dels.png"
                            style="width:20px;height:20px;"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          class="ant-pagination ant-table-pagination"
          unselectable="unselectable"
        >
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-prev"
            tabindex="0"
            title="上一页"
          >
            <a />
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a>
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-next"
            tabindex="0"
            title="下一页"
          >
            <a />
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
